controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -64,6 +65,27 @@ func (in *KubeadmControlPlane) Default() {
 	if !strings.HasPrefix(in.Spec.Version, "v") {
 		in.Spec.Version = "v" + in.Spec.Version
 	}
+
+	ios1 := intstr.FromInt(1)
+	ios0 := intstr.FromInt(0)
+
+	if in.Spec.RolloutStrategy == nil {
+		in.Spec.RolloutStrategy = &RolloutStrategy{}
+	}
+
+	// Enforce RollingUpdate strategy and default MaxSurge and MaxUnavailable if not set.
+	if in.Spec.RolloutStrategy != nil {
+		if in.Spec.RolloutStrategy.Type == RolloutStrategyType("") {
+			in.Spec.RolloutStrategy.Type = RollingUpdateStrategyType
+		}
+
+		if in.Spec.RolloutStrategy.RollingUpdate == nil {
+			in.Spec.RolloutStrategy.RollingUpdate = &RollingUpdate{}
+		}
+
+		in.Spec.RolloutStrategy.RollingUpdate.MaxSurge = intstr.ValueOrDefault(in.Spec.RolloutStrategy.RollingUpdate.MaxSurge, ios1)
+		in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable = intstr.ValueOrDefault(in.Spec.RolloutStrategy.RollingUpdate.MaxUnavailable, ios0)
```

Comment:
Move these inside the `in.Spec.RolloutStrategy == RollingUpdateStrategyType` condition